mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala

```diff
@@ -197,8 +238,86 @@ class LocalLDAModel private[clustering] (
   // TODO:
   // override def topicDistributions(documents: RDD[(Long, Vector)]): RDD[(Long, Vector)] = ???
 
+  /**
+   * Calculate and return per-word likelihood bound, using the `batch` of
+   * documents as evaluation corpus.
+   */
+  // TODO: calcualte logPerplexity over training set online during training, reusing gammad instead
+  // of performing variational inference again in [[bound()]]
+  def logPerplexity(
+      batch: RDD[(Long, Vector)],
+      totalDocs: Long): Double = {
```

Comment:
@jkbradley I think that's happening on L263 actually.

My previous comment was incorrect; log perplexity is -log p(docs) / numWords, so just as you said it is normalized by sample size (i.e. number of words in documents).

I will add this to the docs, but how do you feel about keeping the name since the defn of perplexity accounts for normalizing by num words?